### PR TITLE
Add auth BE to subs

### PR DESCRIPTION
**Pull request by marcosholgado**
Task/Issue URL: https://app.asana.com/0/488551667048375/1205077126622549/f 

### Description
This PR adds auth to subs

### Steps to test this PR
See task

### CODE REVIEW

**Comment by marcosholgado** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3508** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3508" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3508?utm_source=stack-comment).

**Review by karlenDimla — APPROVED** [review]
LGTM. Also tested and worked as expected.

File: subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    private suspend fun createAccount(): CreateAccountResponse {
+        val response = authService.createAccount()
+        authDataStore.token = response.authToken
+        _isSignedIn.emit(isUserAuthenticated())
+        return response
+    }
+
+    private fun parseError(e: HttpException): ResponseError? {
+        return try {
+            val error = Moshi.Builder().build().adapter(ResponseError::class.java).fromJson(e.response()?.errorBody()?.string().orEmpty())
```
[the remark this answers is not in the document]
Comment (marcosholgado) [reply subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/SubscriptionsManager.kt]: Done!

File: subscriptions/subscriptions-impl/src/test/java/com/duckduckgo/subscriptions/impl/RealSubscriptionsManagerTest.kt [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    private val context: Context = mock()
+
+    @Before
+    fun before() {
+        whenever(context.packageName).thenReturn("packageName")
+    }
+
+    @Test
+    fun whenGetExternalIdIfUserNotAuthenticatedAndNotPurchaseStoredThenCreateAccount() = runTest {
+        givenPurchaseStored()
```
[the remark this answers is not in the document]
Comment (marcosholgado) [reply subscriptions/subscriptions-impl/src/test/java/com/duckduckgo/subscriptions/impl/RealSubscriptionsManagerTest.kt]: Correct!

**Pull request merged**